src/types/object.js

```diff
@@ -14,6 +14,15 @@ export default context => {
   }
   const prefix = context.key ? `${context.key}.` : ''
   if (context.def.keys) {
+    // Ensure strict key match
+    if (!context.def.hasOwnProperty('strict') || context.def.strict) {
+      _.forOwn(context.value, (val, key) => {
+        if (!context.def.keys[key]) {
+          context.errors.push({ key, val, message: `'${key}' is not an allowed property` })
```

Comment:
It might make sense to use `context.fail` here. Pedantic, but that way if we ever change the structure of an error object, we only have to do it in one place.